From 2bc0699fabc770e2e79b689685cd1b3b837ead8d Mon Sep 17 00:00:00 2001
From: Yanan Yang <yanan.yang@nxp.com>
Date: Fri, 16 May 2025 16:31:00 +0900
Subject: [PATCH 16/56] LF-15340: arm64: dts: imx8mp-frdm: add
 imx8mp-frdm-8mic-revE.dts

Add 8mic-revE support for FRDM-iMX8MP

Signed-off-by: Yanan Yang <yanan.yang@nxp.com>
Reviewed-by: Chancel Liu <chancel.liu@nxp.com>
---
 arch/arm64/boot/dts/freescale/Makefile        |   1 +
 .../dts/freescale/imx8mp-frdm-8mic-revE.dts   | 207 ++++++++++++++++++
 2 files changed, 208 insertions(+)
 create mode 100644 arch/arm64/boot/dts/freescale/imx8mp-frdm-8mic-revE.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index 1fb6b079ea0a..d28dc95709da 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -258,6 +258,7 @@ dtb-$(CONFIG_ARCH_MXC) += imx8mp-evk-revb4-rpmsg-lpv.dtb
 
 dtb-$(CONFIG_ARCH_MXC) += imx8mp-frdm.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8mp-frdm-ap1302-dual.dtb
+dtb-$(CONFIG_ARCH_MXC) += imx8mp-frdm-8mic-revE.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8mp-frdm-boe-wuxga-lvds0-panel.dtb imx8mp-frdm-boe-wuxga-lvds1-panel.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8mp-frdm-os08a20.dtb imx8mp-frdm-os08a20-dual.dtb
 
diff --git a/arch/arm64/boot/dts/freescale/imx8mp-frdm-8mic-revE.dts b/arch/arm64/boot/dts/freescale/imx8mp-frdm-8mic-revE.dts
new file mode 100644
index 000000000000..40a5b3b709e5
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx8mp-frdm-8mic-revE.dts
@@ -0,0 +1,207 @@
+// SPDX-License-Identifier: (GPL-2.0 OR MIT)
+/*
+ * Copyright 2025 NXP
+ */
+
+#include "imx8mp-frdm.dts"
+
+/ {
+	mic_leds {
+		compatible = "gpio-leds";
+		mic0 {
+			label = "mic0";
+			gpios = <&pca9555 5 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+		mic1 {
+			label = "mic1";
+			gpios = <&pca9555 7 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+		mic2 {
+			label = "mic2";
+			gpios = <&pca9555 6 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+		mic3 {
+			label = "mic3";
+			gpios = <&pca9555 2 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+		mic4 {
+			label = "mic4";
+			gpios = <&pca9555 1 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+		mic5 {
+			label = "mic5";
+			gpios = <&pca9555 0 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+		mic6 {
+			label = "mic6";
+			gpios = <&pca9555 4 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+		mic7 {
+			label = "mic7";
+			gpios = <&pca9555 3 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+	};
+
+	sw_keys {
+		compatible = "gpio-keys";
+
+		sw4: volume_down {
+			label = "Volume Down";
+			gpios = <&pca9555 15 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_VOLUMEDOWN>;
+			interrupt-parent = <&pca9555>;
+			interrupts = <15 IRQ_TYPE_LEVEL_LOW>;
+		};
+
+		sw3: volume_up {
+			label = "Volume Up";
+			gpios = <&pca9555 14 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_VOLUMEUP>;
+			interrupt-parent = <&pca9555>;
+			interrupts = <14 IRQ_TYPE_LEVEL_LOW>;
+		};
+
+		sw2: volume_mute {
+			label = "Volume Mute";
+			gpios = <&pca9555 13 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_MUTE>;
+			interrupt-parent = <&pca9555>;
+			interrupts = <13 IRQ_TYPE_LEVEL_LOW>;
+		};
+
+		sw1: key_act {
+			label = "Key Act";
+			gpios = <&pca9555 12 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_F9>;
+			interrupt-parent = <&pca9555>;
+			interrupts = <12 IRQ_TYPE_LEVEL_LOW>;
+		};
+	};
+
+	reg_vddext_3v3: regulator-vddext {
+		compatible = "regulator-fixed";
+		regulator-name = "VDDEXT_3V3";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+	};
+};
+
+&iomuxc {
+	pinctrl_swpdm_mute_irq: swpdm_mute_grp {
+		fsl,pins = <
+			MX8MP_IOMUXC_ECSPI1_MISO__GPIO5_IO08	0x19
+		>;
+	};
+
+	pinctrl_pushbutton_irq: pushbutton_grp {
+		fsl,pins = <
+			MX8MP_IOMUXC_ECSPI1_SS0__GPIO5_IO09 0xb0
+		>;
+	};
+};
+
+&i2c3 {
+	pca9555: gpio@21 {
+		compatible = "nxp,pca9555";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_pushbutton_irq>;
+		reg = <0x21>;
+		gpio-controller;
+		#gpio-cells = <2>;
+		interrupt-controller;
+		#interrupt-cells = <2>;
+		interrupt-parent = <&gpio5>;
+		interrupts = <9 IRQ_TYPE_LEVEL_LOW>;
+		vcc-supply = <&reg_vddext_3v3>;
+	};
+
+	pca9955btw: pca9955btw@7 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		compatible = "nxp,pca9955b";
+		reg = <0x07>;
+		led@0 {
+			label = "green0";
+			linux,default-trigger = "none";
+			reg = <0>;
+		};
+
+		led@1 {
+			label = "blue0";
+			linux,default-trigger = "none";
+			reg = <1>;
+		};
+
+		led@2 {
+			label = "red0";
+			linux,default-trigger = "none";
+			reg = <2>;
+		};
+
+		led@3 {
+			label = "green1";
+			linux,default-trigger = "none";
+			reg = <3>;
+		};
+
+		led@4 {
+			label = "blue1";
+			linux,default-trigger = "none";
+			reg = <4>;
+		};
+
+		led@5 {
+			label = "red1";
+			linux,default-trigger = "none";
+			reg = <5>;
+		};
+
+		led@6 {
+			label = "green2";
+			linux,default-trigger = "none";
+			reg = <6>;
+		};
+
+		led@7 {
+			label = "blue2";
+			linux,default-trigger = "none";
+			reg = <7>;
+		};
+
+		led@8 {
+			label = "red2";
+			linux,default-trigger = "none";
+			reg = <8>;
+		};
+
+		led@9 {
+			label = "green3";
+			linux,default-trigger = "none";
+			reg = <9>;
+		};
+
+		led@10 {
+			label = "blue3";
+			linux,default-trigger = "none";
+			reg = <10>;
+		};
+
+		led@11 {
+			label = "red3";
+			linux,default-trigger = "none";
+			reg = <11>;
+		};
+	};
+};
+
+&uart3 {
+	status = "disabled";
+};
-- 
2.34.1

